File: java/src/DirectionWest.java

public class DirectionWest implements IDirection
{
	public IDirection left()
	{
		return new DirectionSouth();
	}
	
	public IDirection right()
	{
		return new DirectionNorth();
	}
	
	public Position move(Position position)
	{
		return position;
	}

	public String name()
	{
		return "WEST";
	}
}

File: ruby/src/DirectionWest.rb
require_relative "IDirection.rb"
require_relative "DirectionNorth.rb"
require_relative "DirectionSouth.rb"

class DirectionWest < IDirection
  
  def name()
    return "WEST"
  end
  
  def left()
    return DirectionSouth.new()
  end
  
  def right()
    return DirectionNorth.new()
  end
  
  def move(position)
    position.x_position = position.x_position - 1
    return position
  end
end
